consensus/src/chained_bft/block_storage/block_store.rs

```diff
@@ -55,22 +57,27 @@ pub enum NeedFetchResult {
 ///             | -> C1
 ///             | -------> C2
 ///             | -------------> D3
-pub struct BlockStore<T> {
-    inner: Arc<RwLock<BlockTree<T>>>,
-    validator_signer: ValidatorSigner<Ed25519PrivateKey>,
-    state_computer: Arc<dyn StateComputer<Payload = T>>,
+pub struct BlockStore<T, Sig: Signature> {
```

Comment:
can you rename `T` and `Signature` as `Transaction` and `SignatureType` :D?

Reply:
- renaming `T` : I'll leave it to more interested parties
- renaming `Sig`: why not :)